### DISCUSSION TITLE
Measure Properties dialog won't apply to a range of selected measures

### DISCUSSION THREAD

**Discussion by scorster** [musescore/MuseScore#16286] [General]
In MS3, the Measure Properties dialog silently applies property changes only to the last measure of the existing selection.

MS4 does the same thing, but *ALSO* changes the user's multi-measure selection (made prior to opening Measure Properties) to the last measure of the selection.

Measure Properties dialog is a great way to fix certain types of metric corruption—but currently the user can only change properties one measure at a time because, as stated, property changes apply only to the last measure of the existing selection.
 
I recommend/request that Measure Properties operate on all measures in the selection. MuseScore can identify mixed values by displaying them in gray.

Forum discussion here:

https://musescore.org/en/node/344588#comment-1173080

scorster

**Comment by SteveBlower** [comment]
Seems to make for more consistent behaviour, but ... 

consider the case where you select a range of measures and reduce the actual duration, say from 4/4 to 3/4,  Currently Musescore will remove the last beat from the one measure that it acts upon.  If that beat is occupied by a note that starts on an earlier beat - e.g. a half note on beat 3 - then that note is currently deleted entirely, leaving a rest on beat 3.   How do you propose to handle that action when a range of measures is selected?  Will users always want the last beat of every measure in the selection to be removed?  Or even want the same beat to be removed?

The devil is, as ever, in the detail.

**Reply by scorster** [reply]
@ Steve Blower wrote > *How do you propose to handle that action when a range of measures is selected? Will users always want the last beat of every measure in the selection to be removed? Or even want the same beat to be removed?*

Either approach would be fine, as long as the documentation clearly explains the result. But for consistency I recommend what MuseScore currently does on a single measure.

If "actual duration" has varying values across a selected range of measures MuseScore could:

1) disable "actual duration" in the dialog
2) post an alert (with OK or Cancel options) explaining that it ignore "actual duration" when processing the selection 3) do so on OK
